gordon/interfaces.py

```diff
@@ -53,16 +53,29 @@ def update_phase(new_phase):
 
 
 class IGenericPlugin(Interface):
-    """**Do not** implement this interface directly.
+    """Plugin which is associated with a phase.
 
-    Use :py:class:`gordon.interfaces.IEventConsumerClient`,
-    :py:class:`gordon.interfaces.IEnricherClient`,
-    or :py:class:`gordon.interfaces.IPublisherClient` instead.
+    **Do not** implement this interface directly.
+
+    Use :py:class:`gordon.interfaces.IRunnable`,
+    or :py:class:`gordon.interfaces.IMessageHandler` instead.
     """
     phase = Attribute('Plugin phase')
 
-    def __init__(config, success_channel, error_channel, metrics=None):
-        """Initialize an EventClient object.
+    async def shutdown():
+        """Gracefully shutdown plugin."""
+
+
+class IRunnable(IGenericPlugin):
+    """Plugin for ingesting push/pull events for Gordon to process.
+
+    The plugin also has the ability to send :py:class:`gordon.interfaces
+    EventMessage` objects to both success and error channels. At least
+    one runnable plugin is required to run Gordon.
+    """
+
+    def __init__(config, success_channel, error_channel, metrics, **kwargs):
+        """Initialize an runnable.
```

Comment:
`an` -> `a`; also  probably want to say a runnable _what_.

Reply:
Will fix!